apis/externalsecrets/v1beta1/externalsecret_validator_test.go

```diff
@@ -122,6 +122,21 @@ either data or dataFrom should be specified`,
 				},
 			},
 		},
+		{
```

Comment:
Add one when there are mutliple matching secrets and some that aren't duplicates please :)

Reply:
Added :)

Reply:
ah, meant when there are multiple duplicates, like:

```go

					Data: []ExternalSecretData{
						{SecretKey: "SERVICE_NAME"},
						{SecretKey: "SERVICE_NAME"},
						{SecretKey: "SERVICE_NAME-2"},
						{SecretKey: "SERVICE_NAME-2"},
						{SecretKey: "NOT_DUPLICATE"},
					},
```

This way we can see that it continued and can list multiple problems nicely.

Reply:
Can you review the error ? 
Test cases return same errors but not match.

Reply:
It's joined by new line not `,`. So:

```go
expectedErr: "duplicate secretKey found: SERVICE_NAME\nduplicate secretKey found: SERVICE_NAME-2",
```

Also, please fix the now failing sonarcloud error: https://sonarcloud.io/project/issues?resolved=false&pullRequest=3003&id=external-secrets_external-secrets&open=AYzjXXGiDU-oqzasTv1H

Basically, the function is now too complex. :) Do something like:

```go

func validateDuplicateKeys(es *ExternalSecret, errs error) error {
	if es.Spec.Target.DeletionPolicy == DeletionPolicyRetain {
		seenKeys := make(map[string]struct{})
		for _, data := range es.Spec.Data {
			secretKey := data.SecretKey
			if _, exists := seenKeys[secretKey]; exists {
				errs = errors.Join(errs, fmt.Errorf("duplicate secretKey found: %s", secretKey))
			}
			seenKeys[secretKey] = struct{}{}
		}
	}
	return errs
}

```

And then join if errors is not nil. That should be enough.

Reply:
Done :) @Skarlso